presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -132,7 +131,8 @@ public static long currentTimestamp(ConnectorSession session)
     @SqlType(StandardTypes.TIMESTAMP)
     public static long localTimestamp(ConnectorSession session)
     {
-        return session.getStartTime();
+        ISOChronology localChronology = getChronology(session.getTimeZoneKey());
+        return localChronology.getZone().convertUTCToLocal(session.getStartTime());
```

Comment:
I hope you're aware that `testLocalTimestamp` doesn't pass at this commit. I assume your strategy is _not_ to be green at every step.